### PR TITLE
[MM-40590] Fix inviting a guest guest to a channel where he's a member.

### PR DESCRIPTION
**Pull request by JulienTant**
#### Summary
This PR fixes the bug that caused inviting a guest to a channel where he's already a member to show a "Success" message instead of a message saying the guest is already present.

Unrelated to the ticket but fixed here: during the development, I noticed that if the inviter is not in a channel when trying to invite a people (e.g. if he is on the /tips page during the onboarding process), then clicking on the "Invite As: Guest" makes the webapp crash. I pushed to fix to use the "Town Square" when there's no active channels and added the tests for it.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-40590

#### Related Pull Requests
N/A

#### Screenshots
N/A

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Review by neallred — APPROVED** [review]
Good find and fixes! If I understand the issue, it is that the channel membership data may or may not be loaded yet, so when members was selected, it didn't necessarily show that some guests already belonged to the channel. This PR changes that by making sure the relevant data is loaded first.

File: components/invitation_modal/add_to_channels.test.tsx [hunk 1]
```diff
@@ -28,15 +30,45 @@ const defaultProps: Props = deepFreeze({
...
+    </IntlProvider>
+);
+
 describe('AddToChannels', () => {
     beforeEach(() => {
         props = defaultProps;
     });
+
+    describe('placeholder selection', () => {
+        it('should use townSqureDisplayName when not in a channel', () => {
```
Comment (neallred) [inline components/invitation_modal/add_to_channels.test.tsx:53]: Nice tests!

**Review by neallred — APPROVED** [review]

**Review by saturninoabril — CHANGES_REQUESTED** [review]
Thanks @JulienTant! Tested and passed. Left minor comments but all good to me.

**Review by saturninoabril — APPROVED** [review]
Thanks @JulienTant. Tested and passed. LGTM.

**Review by stevemudie — APPROVED** [review]
Unable to invite a guest guest twice to the same channel as described in the PR title. 😛

Looks good @JulienTant!

**Pull request merged**